discovery/pkg/openstack/reconciler.go

```diff
@@ -154,7 +154,22 @@ func (r *Reconciler) reconcile() {
 		r.reconcileSvcs(desiredSvcs, currentServices.Items)
 
 		desiredEndpoints := kubeEndpoints(r.BackendName, projectName, loadbalancers, pools)
+		for _, ep := range desiredEndpoints {
+			desiredEndpoints = append(desiredEndpoints, ep)
+		}
 		r.reconcileEndpoints(desiredEndpoints, currentEndpoints.Items)
+
+		// Log upstream services prometheus
+		r.Metrics.DiscovererUpstreamServicesMetric(r.BackendName, projectName, totalUpstreamServices)
+		r.Metrics.DiscovererInvalidServicesMetric(r.BackendName, projectName, totalUpstreamServices-len(loadbalancers))
+		r.Metrics.DiscovererUpstreamEndpointsMetric(r.BackendName, projectName, r.sumEndpoints(desiredEndpoints))
+
+		endpoints, err := r.GimbalKubeClient.CoreV1().Endpoints(projectName).List(metav1.ListOptions{})
+		if err != nil {
+			r.Logger.Error(err)
+		}
+		r.Metrics.DiscovererReplicatedEndpointsMetric(r.BackendName, projectName, r.sumEndpoints(endpoints.Items))
```

Comment:
Is this competing with the metric being set in `sync.Action`? It's unclear to me why we need them in both places. Also seems like we are missing the equivalent for `DiscovererReplicatedServicesMetric`?